### PR TITLE
docs: Update tutor.rst with additional information about contributing in plugins

### PR DESCRIPTION
**Pull request by Abdul-Muqadim-Arbisoft**
Additional information added in contribution section of tutor.rst about contributing in plugins

### CODE REVIEW

File: docs/tutor.rst [hunk 1]
```diff
@@ -100,13 +100,30 @@ Contributing to Tutor
...
+- **Code Formatting**: Format your code, if necessary, using ``make format``.
+- **Commit Messages**: Craft a clear Git commit title and message. Explain the rationale behind your changes, the problem you're addressing, and the chosen solution. Link to relevant forum discussions and detail your use case. We appreciate detailed explanations. Use `conventional commits <https://www.conventionalcommits.org>`__ for the title.
 
-Happy hacking! ☘️
+Plugin-Specific Guidelines
+--------------------------
+
+For plugin contributions, keep the following additional points in mind:
+
+- Do not bump the version number in your contributions. This is handled separately during the release process.
```
Comment (regisb) [inline docs/tutor.rst:115]: This is not a plugin-specific recommendation. It also applies to Tutor core. So I think we should not have a "plugin-specific guidelines" section, as all recommendations are both for Tutor core and plugins.

File: docs/tutor.rst [hunk 2]
```diff
@@ -100,13 +100,30 @@ Contributing to Tutor
...
+- **General Discussion**: For anything beyond obvious bugs, initiate a discussion in the `official Open edX forum <https://discuss.openedx.org>`__. This helps in agreeing on a high-level solution.
+- **Pull Requests**: Open a pull request on the `Tutor repository <https://github.com/overhangio/tutor/pulls>`__ for changes to the Tutor core or the corresponding plugin for plugin-specific changes.
+- **Running Tests**: Ensure that all tests pass by running ``make test``. This applies to both the Tutor core and plugin contributions.
+- **Changelog Entry**: If your PR is in the Tutor core repository or affects plugin users, add a changelog entry by running ``make changelog-entry``. Edit the new file as per the formatting instructions.
+- **Code Formatting**: Format your code, if necessary, using ``make format``.
```
Comment (regisb) [inline docs/tutor.rst:107]: This should be moved inside the "running tests" section. Something along the lines of: "if formatting tests fail, make sure to run `make format`.

File: docs/tutor.rst [hunk 3]
```diff
@@ -100,13 +100,30 @@ Contributing to Tutor
...
-- Write a good Git commit title and message: explain why you are making this change, what problem you are solving and which solution you adopted. Link to the relevant conversation topics in the forums and describe your use case. We *love* long, verbose descriptions :) As for the title, `conventional commits <https://www.conventionalcommits.org>`__ are preferred. Check the repo history!
+- **General Discussion**: For anything beyond obvious bugs, initiate a discussion in the `official Open edX forum <https://discuss.openedx.org>`__. This helps in agreeing on a high-level solution.
+- **Pull Requests**: Open a pull request on the `Tutor repository <https://github.com/overhangio/tutor/pulls>`__ for changes to the Tutor core or the corresponding plugin for plugin-specific changes.
+- **Running Tests**: Ensure that all tests pass by running ``make test``. This applies to both the Tutor core and plugin contributions.
+- **Changelog Entry**: If your PR is in the Tutor core repository or affects plugin users, add a changelog entry by running ``make changelog-entry``. Edit the new file as per the formatting instructions.
```
Comment (regisb) [inline docs/tutor.rst:106]: Note that it is not necessary to create a changelog entry for reformatting or documentation contributions.

File: docs/tutor.rst [hunk 4]
```diff
@@ -100,13 +100,30 @@ Contributing to Tutor
...
 
-Happy hacking! ☘️
+Plugin-Specific Guidelines
+--------------------------
+
+For plugin contributions, keep the following additional points in mind:
+
+- Do not bump the version number in your contributions. This is handled separately during the release process.
+
+Releasing a New Version
```
Comment (regisb) [inline docs/tutor.rst:117]: Less upper-case: "Releasing a new version"

File: docs/tutor.rst [hunk 5]
```diff
@@ -100,13 +100,30 @@ Contributing to Tutor
...
+-----------------------
+
+For releasing a new version :
+
+- **Version Number**: Bump the version number in ``__about__.py``.
```
Comment (regisb) [inline docs/tutor.rst:122]: Add a note linking to the versioning guidelines.

**Review by regisb — CHANGES_REQUESTED** [review]
Thanks a lot for this PR! it's a good start.

File: docs/tutor.rst [hunk 6]
```diff
@@ -98,13 +98,24 @@ An optional BRANCH suffix may be appended to the release name to indicate that e
...
+- **Pull Requests**: For changes to Tutor core or plugin-specific modifications, open a pull request on the `Tutor repository <https://github.com/overhangio/tutor/pulls>`__.
+- **Running Tests and Code Formatting**:
+  - Ensure all tests pass by running ``make test``. This is mandatory for both Tutor core and plugin contributions.
+  - If formatting tests fail, correct your code format using ``make format``.
+- **Changelog Entry**: Create a changelog entry for significant changes (excluding reformatting or documentation) by running ``make changelog-entry``. Edit the newly created file following the given formatting instructions. This applies to both Tutor core and plugin changes.
+- **Commit Messages**: Write clear Git commit titles and messages. Detail the rationale for your changes, the issue being addressed, and your solution. Include links to relevant forum discussions and describe your use case. Detailed explanations are valuable. For commit titles, follow `conventional commits <https://www.conventionalcommits.org>`__ guidelines.
```
Comment (regisb) [inline docs/tutor.rst:109]: "If your pull request addresses an existing GitHub issue, add 'Close #XXX' to your commit message, where XXX is the issue number."

File: docs/tutor.rst [hunk 7]
```diff
@@ -98,13 +98,24 @@ An optional BRANCH suffix may be appended to the release name to indicate that e
...
-- Make sure that all tests pass by running ``make test`` (see above).
-- If your PR is in the Tutor core repository, add a changelog entry by running ``make changelog-entry``. Edit the new file and follow the formatting instructions that it contains.
-- Write a good Git commit title and message: explain why you are making this change, what problem you are solving and which solution you adopted. Link to the relevant conversation topics in the forums and describe your use case. We *love* long, verbose descriptions :) As for the title, `conventional commits <https://www.conventionalcommits.org>`__ are preferred. Check the repo history!
+- **General Discussion**: Before addressing anything other than clear-cut bugs, start a discussion on the `official Open edX forum <https://discuss.openedx.org>`__. This facilitates reaching a consensus on a high-level solution.
+- **Pull Requests**: For changes to Tutor core or plugin-specific modifications, open a pull request on the `Tutor repository <https://github.com/overhangio/tutor/pulls>`__.
```
Comment (regisb) [inline docs/tutor.rst:104]: ... or the corresponding plugin repository.

File: docs/tutor.rst (hunk 6 above)
Comment (Abdul-Muqadim-Arbisoft) [reply docs/tutor.rst]: Added

File: docs/tutor.rst (hunk 7 above)
Comment (Abdul-Muqadim-Arbisoft) [reply docs/tutor.rst]: Corrected

**Review by regisb — APPROVED** [review]
Awesome!

**Pull request merged**